prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctions.kt

```diff
@@ -353,6 +367,22 @@ object CustomFHIRFunctions : FhirPathFunctions {
         return if (type != null) mutableListOf(StringType(type)) else mutableListOf()
     }
 
+    fun deidentifyHumanName(focus: MutableList<Base>, parameters: MutableList<MutableList<Base>>?): MutableList<Base> {
```

Comment:
I can't recall if you went over this during your 17th, how did we arrive at just overwriting the `given` and `family` fields?

Looking at our [HumanName](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/metadata/HL7/catchall/hl7/datatypes/XPN/HumanName.yml) mappings that [patient](https://github.com/CDCgov/prime-reportstream/blob/eb84e75607c7e89bb8416de58951b73314a1b8ce/prime-router/metadata/HL7/catchall/hl7/segments/PID/Patient.yml#L35-L50) uses, we would want to also overwrite `_family`. Less important but still feels like potentially PII is `period` as that could reveal a birthdate or the date of a name change.


Not really sure what our ultimate goal with deidentification is, but it seems like there's quite a lot of PII not covered by HumanName datatypes. There's a few other name-realted PID fields that seem like they should be dealt with in a similar way if they want the data deitentified (PID.6 and PID.9 for example, well I guess PID.6 is the same data format so it wouldn't be a huge issue). But I think those are pretty firmly out of scope for this.

Reply:
My idea is that this was more launching point for the engagement team to add to as it becomes clearer what deidentification means.  I'm on the same page that its not super clear, but I think thats a topic for the engagement team to work out

Reply:
Makes sense to me!